android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenManager.java

```diff
@@ -376,16 +379,34 @@ private void downloadDeviceIcon(final String url){
 			@Override
 			public void run(){
 				try{
-					deviceLogo = AndroidTools.downloadImage(url);
+					if(mLockScreenDeviceIconManager.updateCachedImage(url)) {
+						Log.d(TAG, "URL: " + url);
+						Log.d(TAG, "Image Update Needed");
+						deviceLogo = AndroidTools.downloadImage(url);
+						mLockScreenDeviceIconManager.saveFileToCache(deviceLogo, url);
+					} else {
+						Log.d(TAG, "Image Is Up To Date");
```

Comment:
it is better to use DebugTool for logging D/I messages

Reply:
changed to. DebugTool